doc/usage.rst

```diff
@@ -89,6 +89,30 @@ Or in Windows, use the little-known BAT file line continuation::
         myscript.spec
 
 
+Running PyInstaller with Python optimizations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. Note::
+
+	You should be aware of the ``PYTHONOPTIMIZE`` mechanism when using
+	this option. Using ``-O``, ``__debug__`` is set to ``False`` and
+	``assert`` statements are not executed. The ``-OO`` option
+	additionally omits docstrings. If your code relies on any of these
+	features, your program may break or have unexpected behavior.
```

Comment:
We should add something like "Using this feature/option/thing effects not only your main script but *all* modules compiled by |PyInstaller|"